bluemira/equilibria/coils.py

```diff
@@ -723,1492 +743,1226 @@ def F(self, eqcoil):  # noqa :N802
+        self.x = new_position[:, 0]
+        self.z = new_position[:, 1]
 
-    def control_Bx(self, x, z):
+    def adjust_position(self, d_xz: __ITERABLE_FLOAT):
         """
-        Returns a list of control responses for Bx at the given (x, z)
-        location(s)
+        Adjust position of each coil
         """
-        return sum(super().control_Bx(x, z))
+        d_xz = np.atleast_2d(d_xz)
+        self.position = np.stack(
+            [self.x.ravel() + d_xz[:, 0], self.z.ravel() + d_xz[:, 1]], axis=1
+        )
 
-    def control_Bz(self, x, z):
+    @property
+    def x(self) -> tuple:
         """
-        Returns a list of control responses for Bz at the given (x, z)
-        location(s)
+        Get x coordinate of each coil
         """
-        return sum(super().control_Bz(x, z))
+        _x = self._x[:]
+        _x.flags.writeable = False
+        return _x
 
-    def control_psi(self, x, z):
+    @x.setter
+    def x(self, new_x: __ITERABLE_FLOAT) -> None:
         """
-        Returns a list of control responses for psi at the given (x, z)
-        location(s)
+        Set x coordinate of each coil
         """
-        return sum(super().control_psi(x, z))
+        self._x[:] = np.atleast_2d(new_x.T).T
+        self._sizer(self)
 
-    def control_F(self, coil):
+    @property
+    def z(self) -> tuple:
         """
-        Returns a list of control responses for F at the given (x, z)
-        location(s)
+        Get z coordinate of each coil
         """
-        return np.sum(super().control_F(coil), axis=0)
+        _z = self._z[:]
+        _z.flags.writeable = False
+        return _z
 
-    def F(self, eqcoil):
+    @z.setter
+    def z(self, new_z: __ITERABLE_FLOAT) -> None:
         """
-        Get the sum of the forces on a coil from the Circuit.
+        Set z coordinate of each coil
         """
-        return np.sum(super().F(eqcoil), axis=0)
+        self._z[:] = np.atleast_2d(new_z.T).T
+        self._sizer(self)
 
     @property
-    def n_control(self):
+    def dx(self) -> tuple:
         """
-        The length of the controls.
+        Get dx coordinate of each coil
         """
-        return 1
+        _dx = self._dx[:]
+        _dx.flags.writeable = False
+        return _dx
 
-    @property
-    def n_constraints(self):
+    @dx.setter
+    def dx(self, new_dx: __ITERABLE_FLOAT) -> None:
         """
-        The length of the constraints.
+        Set dx coordinate of each coil
         """
-        return len(self.coils)
+        self._dx[:] = np.atleast_2d(new_dx.T).T
+        self._sizer(self)
 
-    def make_size(self, current=None):
+    @property
+    def dz(self) -> np.ndarray:
         """
-        Set the size of the coils in the Circuit.
+        Get dz coordinate of each coil
         """
-        for coil in self.coils.values():
-            coil.make_size(current=current)
+        _dz = self._dz[:]
+        _dz.flags.writeable = False
+        return _dz
```

Comment:
Woah.. what's this a read-only view? Is that not safer as a copy? I'm not sure what the reasons for this are, but I think a comment would be good!